dev/com.ibm.ws.webcontainer.security/test/com/ibm/ws/webcontainer/security/internal/AuthenticateApiTest.java

```diff
@@ -414,6 +425,112 @@ public void testLoginSuccess() throws Exception {
+        BasicAuthAuthenticator basicAuthAuthenticator = new BasicAuthAuthenticator(authnService, userRegistry, ssoCookieHelper, config);
+        
+        try {
+            authApi.login(req, resp, user, password, config, basicAuthAuthenticator);
+        } catch (Exception e) {
+            boolean foundException = false;
+            if(e instanceof com.ibm.websphere.security.web.PasswordExpiredException){
+                foundException = true;
+            }
+            assertEquals(e.getMessage(), true, foundException);
+        }
+    }
+    /**
+     */
+    @Test
+    public void userRevoked () throws Exception {
+      
+        final CollaboratorUtils cu = mock.mock(CollaboratorUtils.class);
+        
+        final String jaasEntryName = JaasLoginConfigConstants.SYSTEM_WEB_INBOUND;
+        subjectManager.clearSubjects();
+        final Subject subject = createAuthenticatedSubject();
+        final AuthenticationData authenticationData = createAuthenticationData(user, password);
+        mock.checking(new Expectations() {
+            {
+                one(config).getLogoutOnHttpSessionExpire();
+                will(returnValue(false));
+                one(req).getRequestedSessionId();
+                will(returnValue("abc"));
+                one(req).isRequestedSessionIdValid();
+                will(returnValue(false));
+                allowing(authnService).authenticate(with(equal(jaasEntryName)), with(matchingAuthenticationData(authenticationData)), with(equal((Subject) null)));
+                will(throwException(new com.ibm.ws.security.authentication.UserRevokedException("authn failed")));
+                one(cu).getUserRegistryRealm(securityServiceRef);
+                will(returnValue("joe"));
+                allowing(securityServiceRef).getService();
+                will(returnValue(securityService));
+                allowing(securityService).getAuthenticationService();
+                will(returnValue(authnService));
+                allowing(authnService).getAuthCacheService();
+                allowing(req).getRemoteUser();
+                allowing(req).getUserPrincipal();
+                allowing(ssoCookieHelper).getSSOCookiename();
+                allowing(req).getCookies();
+                allowing(req).getServletContext();
+                will(returnValue(mockServletContext));
+                allowing(mockServletContext).getAttribute(with(any(String.class)));
+                will(returnValue(null));  
+            }
+        });
+        
+        AuthenticateApi authApi = new AuthenticateApi(ssoCookieHelper, securityServiceRef, cu, null, new ConcurrentServiceReferenceMap<String, UnprotectedResourceService>("unprotectedResourceService"));
+        BasicAuthAuthenticator basicAuthAuthenticator = new BasicAuthAuthenticator(authnService, userRegistry, ssoCookieHelper, config);
+        
+        try {
+            authApi.login(req, resp, user, password, config, basicAuthAuthenticator);
```

Comment:
Same comment here about failing if we don't throw an exception.  I think there is a junit `fail(message)` method that should work.